dot/rpc/subscription/listeners.go

```diff
@@ -286,3 +286,35 @@ func (l *RuntimeVersionListener) Listen() {
 // Stop to runtimeVersionListener not implemented yet because the listener
 // does not need to be stoped
 func (l *RuntimeVersionListener) Stop() {}
+
+// GrandpaJustificationListener struct has the finalisedCh and the context to stop the goroutines
+type GrandpaJustificationListener struct {
+	cancel context.CancelFunc
+	ctx    context.Context
+	wsconn *WSConn
+	subID  uint
+
+	finalisedChID byte
+	finalisedCh   chan *types.FinalisationInfo
+}
+
+const grandpaJustifications = "grandpa_justifications"
+
+// Listen will start goroutines that listen to the finaised blocks
+func (g *GrandpaJustificationListener) Listen() {
+	// listen for finalised headers
+	go func() {
+		for {
+			select {
+			case info := <-g.finalisedCh:
+				hash := info.Header.Hash().String()
+				g.wsconn.safeSend(newSubscriptionResponse(grandpaJustifications, g.subID, hash))
+			case <-g.ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+// Stop will cancel all the goroutines that are executing
+func (g *GrandpaJustificationListener) Stop() { g.cancel() }
```

Comment:
probably should ensure that `Listen()` has returned, and or return an error after timeout

Reply:
Done!